commit 201eecccc01177e7f87d25fba9eec8c0756821c4
Author: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date:   Fri Nov 20 14:01:26 2015 -0600

    UBUNTU: SAUCE: Clear Linux: init: do_mounts: recreate /dev/root
    
    Rootfs shows as is mounted in /dev/root, but this devices is not present in
    /dev directory.
    
    Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/init/do_mounts.c b/init/do_mounts.c
index f6d4dd764a52..9fd3875897df 100644
--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -549,6 +549,7 @@ void __init mount_root(void)
 void __init prepare_namespace(void)
 {
 	int is_floppy;
+	int err;
 
 	if (root_delay) {
 		printk(KERN_INFO "Waiting %d sec before mounting root device...\n",
@@ -602,6 +603,13 @@ void __init prepare_namespace(void)
 	devtmpfs_mount("dev");
 	sys_mount(".", "/", NULL, MS_MOVE, NULL);
 	sys_chroot(".");
+#ifdef CONFIG_BLOCK
+	/* recreate the /dev/root */
+	err = create_dev("/dev/root", ROOT_DEV);
+
+	if (err < 0)
+		pr_emerg("Failed to create /dev/root: %d\n", err);
+#endif
 }
 
 static bool is_tmpfs;
